examples/fluids/qfunctions/newtonian.h

```diff
@@ -348,6 +359,16 @@ CEED_QFUNCTION_HELPER int IJacobian_Newtonian(void *ctx, CeedInt Q, const CeedSc
     UnpackState_U(ds.U, dU);
     for (int j = 0; j < 5; j++) v[j][i] = wdetJ * (context->ijacobian_time_shift * dU[j] - dbody_force[j]);
 
+    if (context->idl_enable) {
+      const CeedScalar sigma      = LinearRampCoefficient(1 / context->idl_decay_time, context->idl_length, context->idl_start, x_i[0]);
+      const CeedScalar damp_dY[5] = {sigma * ds.Y.pressure, 0, 0, 0, 0};
+      CeedScalar       dx_i[3]    = {0};
+      State            damp_ds    = StateFromY_fwd(context, s, damp_dY, x_i, dx_i);
+      v[0][i] += wdetJ * damp_ds.U.density;
+      for (int j = 0; j < 3; j++) v[j + 1][i] += wdetJ * damp_ds.U.momentum[j];
+      v[4][i] += wdetJ * damp_ds.U.E_total;
+    }
```

Comment:
Should we move this into a helper since it's near-identical to above (and we'll perhaps need it when we have RANS residuals so best to keep qfunctions slim).

Reply:
I think I've done what you suggested, but let me know if I misunderstood.